name.abuchen.portfolio/src/name/abuchen/portfolio/util/WebAccess.java

```diff
@@ -5,78 +5,214 @@
+ *              userAgent :: Assigns User-Agent value. Default is determined for OS platform Windows, Apple and Linux.
+ *              
+ *   ignoreContentType​(java.lang.Boolean ignoreContentType)
+ *         > Parameters
+ *              ignoreContentType :: Configures the request to ignore the Content-Type of the response.
+ *                                   Default (false) handled mime types:
+ *                                   text/plain, application/json, application/xhtml+xml, application/xml, text/html, text/plain, text/xml
+ *
+ *
+ * Example
+ * 
+ *   Target URL
+ *          http://example.com/path/page.html?parameter=value
+ * 
+ *  String html = new WebAccess("example.com", "/path/page.html")
+ *                       .withScheme("http")
+ *                       .addParameter("parameter", "value")
+ *                       .addHeader("Content-Type", "application/json;chartset=UTF-8")
+ *                       .addHeader("X-Response", "daily")
+ *                       .addUserAgent("Mozilla/1.0N (Windows)")
+ *                       .ignoreContentType(true)
+ *                       .get();
+ */
+
 public class WebAccess
 {
     public static final RequestConfig defaultRequestConfig = RequestConfig.custom().setSocketTimeout(20000)
-                    .setConnectTimeout(2000).setConnectionRequestTimeout(20000).build();
+                    .setConnectTimeout(2000).setConnectionRequestTimeout(20000).setCookieSpec(CookieSpecs.STANDARD)
+                    .build();
+
+    private static ArrayList<String> filterContentType = new ArrayList<>();
+
+    static
+    {
+        filterContentType.add(ContentType.DEFAULT_TEXT.getMimeType().toString());
+        filterContentType.add(ContentType.APPLICATION_JSON.getMimeType().toString());
+        filterContentType.add(ContentType.APPLICATION_XHTML_XML.getMimeType().toString());
+        filterContentType.add(ContentType.APPLICATION_XML.getMimeType().toString());
+        filterContentType.add(ContentType.TEXT_HTML.getMimeType().toString());
+        filterContentType.add(ContentType.TEXT_PLAIN.getMimeType().toString());
+        filterContentType.add(ContentType.TEXT_XML.getMimeType().toString());
+    }
 
+    private String url;
     private String scheme = "https"; //$NON-NLS-1$
     private String host;
     private String path;
+    private String userAgent = OnlineHelper.getUserAgent();
+    private Boolean ignoreContentType = false;
     private List<Header> headers = new ArrayList<>();
     private List<NameValuePair> parameters = new ArrayList<>();
 
     public WebAccess(String host, String path)
     {
-        this.host = host;
-        this.path = path;
+        if (Objects.nonNull(host) && Objects.nonNull(path))
+        {
+            this.host = host.trim();
+            this.path = path.trim();
+        }
+    }
+
+    public WebAccess(String url)
+    {
+        this.url = Objects.requireNonNull(url.trim(), "PP WebAccess: url is null"); //$NON-NLS-1$
```

Comment:
Hier greifen wir auf ```url``` zu (die #trim Methode) und prüfen anschließend ob das Objekt vielleicht ```null``` ist. Das wird vorher schon mit einer NPE fliegen.

Ich habe es in ```Objects.requireNonNull(url).trim()``` geändert.